pkg/agent/config.go

```diff
@@ -145,6 +152,12 @@ func (c *Config) validate() error {
 	erc.Whenf(ec, c.Metrics.LoadMetricPrefix == "", emptyTmpl, ".metrics.loadMetricPrefix")
 	erc.Whenf(ec, c.Metrics.SecondsBetweenRequests == 0, zeroTmpl, ".metrics.secondsBetweenRequests")
 	erc.Whenf(ec, c.Scaling.RequestTimeoutSeconds == 0, zeroTmpl, ".scaling.requestTimeoutSeconds")
+	if err := c.Scaling.DefaultConfig.Validate(); err != nil {
+		stack := err.(*erc.Stack) //nolint:errorlint // return ty guaranteed by Validate()
+		if err := fun.Observe(context.TODO(), stack.Iterator(), ec.Add); err != nil {
+			panic(fmt.Errorf("unexpected failure Observe-ing errors: %w", err))
+		}
+	}
```

Comment:
oh, you can just `ec.Add()` the error and [it'll do what you want](https://github.com/tychoish/fun/blob/main/erc/errors.go#L33) (append the input stack to the collector's stack.)

Reply:
also, there's `erc.Unwind` which returns an error slice which might be more usable in some situations.

also there's this which might help in  the future: https://github.com/tychoish/fun/blob/main/erc/helpers.go#L245

Reply:
Done